staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -158,6 +159,9 @@ type Store struct {
 	// ExportStrategy implements resource-specific behavior during export,
 	// optional. Exported objects are not decorated.
 	ExportStrategy rest.RESTExportStrategy
+	// TableConvertor is an optional interface for transforming items or lists
+	// of items into tabular output. If unset, the default will be used.
+	TableConvertor rest.TableConvertor
```

Comment:
Barring `ExportStrategy`, the rest of this interface is about managing basic CRUD.  Can you keep the layers separate so we have one layer for CRUD which can be wrapped by something which can do a TableConversion separately?